numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -3214,6 +3214,10 @@ void_arrtype_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
         Py_SET_SIZE((PyVoidScalarObject *)ret, (int) memu);
         ((PyVoidScalarObject *)ret)->descr =
             PyArray_DescrNewFromType(NPY_VOID);
+        if ( ((PyVoidScalarObject *)ret)->descr == NULL) {
+            npy_free_cache(destptr, memu);
+            return NULL;
+        }
         ((PyVoidScalarObject *)ret)->descr->elsize = (int) memu;
         ((PyVoidScalarObject *)ret)->flags = NPY_ARRAY_BEHAVED |
                                              NPY_ARRAY_OWNDATA;
```

Comment:
Annoying, but could you move the static stuff (flags + base) further up.  And then simply call `Py_DECREF(ret)` for cleanup?  The dealloc function can deal with NULL'ed `descr` just fine, but it does need the OWNDATA flag.

Right now, we are not cleaning up the object itself.

Reply:
done